### PR TITLE
Fix mission end screens for challenges

### LINKED ISSUE #12: Duplicate tiles of finished challenges at the end of the mission
Peacock version: 5.4.0
Game: Hitman III (Epic games)

Currently, finished challenges will have duplicate tiles at the end of the mission. This applies to any finished challenges, not only those that were finished during the current run, but also those finished in previous runs.

image

### PR DESCRIPTION
**Pull request by moonysolari**
- Fixes #12 
- Challenge tiles that got "ticked" on a previous mission end screen will not be ticked in succeeding runs.
- Challenge tiles that did not get a chance to be "ticked" on a mission end screen will be ticked the next time a player triggers the mission end screen.
- Adds a `Ticked` field to every `ChallengeProgression` object in the player profile.
- Compatible with previous player profile saves.

### CODE REVIEW

File: components/profileHandler.ts [hunk 1]
```diff
@@ -537,11 +537,13 @@ profileRouter.post(
         }
 
         for (const challenge of challenges) {
+            // TODO: Add actual support for shortcut challenges
```
Comment (AnthonyFuller) [inline components/profileHandler.ts:540]: Care to elaborate?

**Review by AnthonyFuller — APPROVED** [review]
LGTM, just reply to my comment.

File: components/profileHandler.ts (hunk 1 above)
Comment (RDIL) [reply components/profileHandler.ts:540]: Pretty sure this is my to-do from a while back

Comment (moonysolari) [reply components/profileHandler.ts:540]: Yeah it's #16

Comment (AnthonyFuller) [reply components/profileHandler.ts:540]: Ah okay.

**Comment by moonysolari** [comment]
Might be related to #86, as global challenges should also display similar behavior.

**Comment by LennardF1989** [comment]
> Might be related to #86, as global challenges should also display similar behavior.
I reckon my thing at some point is going to combine with the missionEnd of scoreHandler as well (instead of acting as a plugin), so most likely, I will just have to grab "unticked" challenges to decide on the amount of XP to grant the user profile.

**Review by LennardF1989 — APPROVED** [review]

**Comment by AnthonyFuller** [comment]
> I will just have to grab "unticked" challenges to decide on the amount of XP to grant the user profile.

Regarding challenges, XP is given AS SOON as they're unlocked. Like, pretty much instantly.

**Comment by LennardF1989** [comment]
> Regarding challenges, XP is given AS SOON as they're unlocked. Like, pretty much instantly. No waiting for the end screen, during play.

That's fine, but I believe the original game will increment your Mastery XP progress bar in the first missionend as if you didn't have them before yet, no? If so, we also need something like an "original XP" before any challenges were applied. Unless we decide to just not do this in Peacock :P

**Comment by AnthonyFuller** [comment]
> I believe the original game will increment your Mastery XP progress bar in the first missionend as if you didn't have them before yet, no?

It adds the XP onto mastery (and your player level) as soon as you complete the challenge, what you're thinking of is the visual part of it, which yes, occurs the first time after you've got that XP. So we'll likely need to have something like we have for ticked challenges which is a store of how much XP we need to show the player. This idea, of course, needs to be developed.

**Review by RDIL — APPROVED** [review]

**Pull request merged**